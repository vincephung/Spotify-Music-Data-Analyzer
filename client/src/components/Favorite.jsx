import React from "react";
import "bootstrap/dist/css/bootstrap.min.css";
import { Button, Card, ListGroup, Image } from "react-bootstrap";
import './css/favorite.css'
import genreImg from '../pages/img/genre.jpg'

// Component for a user's favorite item (song, genre, album, artist)
const Favorite = ({
    type,
    ranks
}) => {
    return <Card className="topCard" >
        <Card.Header className="topCardHeader">
            <div className="albumImgContainer">
                <Image className="albumImg"
                    thumbnail
                    src={(() => {
                        if (type === 'Genres') {
                            return genreImg
                        }
                        return ranks[0].image
                    })()}
                ></Image>
            </div>
            Top {type}
        </Card.Header>
        <Card.Body className="topCardBody">
            <Card.Title>
                <div className="topItem">
                    <div className="number-one">1 </div>
                    <div className="number-one-item">
                        {ranks[0].loading ? "[loading]" : ranks[0].mediaName} {
                            type === 'Genres' || type === 'Artists' ? <></> :
                                <span style={{ color: "grey" }}>
                                    &nbsp;&nbsp;by {ranks[0].loading ? "[loading]" : ranks[0].creators.join(', ')} &nbsp;
                                </span>
                        }
                    </div>
                </div>
            </Card.Title>
            <ListGroup className="topListGroup" variant="flush" as="ul">
                {ranks.slice(1).map((rank,index)=>{
                    return <ListGroup.Item as="li">
                    <div style={{ textAlign: "Left" }}>
                        <span className="number"> {index+1}</span>
                        &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                        {rank.loading ? "[loading]" : rank.mediaName.trim()}
                        {
                            type === 'Genres' || type === 'Artists' ? <></> :
                                <span style={{ color: "grey" }}>
                                    &nbsp;&nbsp; by {rank.loading ? "[loading]" : rank.creators.join(', ')}
                                </span>
                        }
                    </div>
                </ListGroup.Item>
                })}
            </ListGroup>
            <div style={{
                display: "flex",
                justifyContent: "right",
            }}>
                <Button className="viewAll"
                    variant="primary"
                    href={`/all/${type}`}
                >View All</Button>
            </div>
        </Card.Body>
    </Card>
}


export default Favorite;
