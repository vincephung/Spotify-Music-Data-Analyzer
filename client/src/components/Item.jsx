import React from 'react';
import "./css/item.css";

// Component for the header row
const HeaderRow = ({ type }) => {
    type = type.replace(/\w\S*/g, (w) => (w.replace(/^\w/, (c) => c.toUpperCase())));
    return (
        <>
            <div style={{
                fontSize: "20px",
                fontWeight: "bolder"
            }}>
                {type}
            </div>
        </>
    )
}

const Item = ({
    type,
    rank,
    mediaName,
    creators,
}) => {
    return (
        <div className="allContainer">
            {
                rank === 0 ? <HeaderRow type={type} /> :
                    <>
                        <div className="all">
                            <span className="number">
                                {rank.trim()}
                            </span>
                            &nbsp;&nbsp;{mediaName.trim()}
                            {
                                type === 'genres' || type === 'artists' ? <></> :
                                    <span style={{ color: "grey" }}>
                                        &nbsp;&nbsp;by {creators.join(', ')}
                                    </span>
                            }
                        </div>
                    </>
            }
        </div>
    )
}

export default Item;
