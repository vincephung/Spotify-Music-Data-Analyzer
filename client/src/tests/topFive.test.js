import Favorite from '../components/Favorite';
import { render, unmountComponentAtNode  } from 'react-dom';
import Item from '../components/Item';
import { act } from "react-dom/test-utils";

let container = null;
let loading = [];

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("should render top 5 songs", () => {
  for (let index = 0; index < 5; index++) {
    loading.push({
      loading: true
    });
  }

  act(() => {
    render(<Favorite type="Songs" ranks={loading}/>, container);
  });

  expect(container).toMatchSnapshot(); /* ... gets filled automatically by jest ... */

});