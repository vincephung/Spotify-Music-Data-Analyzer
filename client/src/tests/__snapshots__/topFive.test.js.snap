// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render top 5 songs 1`] = `
<div>
  <div
    class="topCard card"
  >
    <div
      class="topCardHeader card-header"
    >
      <div
        class="albumImgContainer"
      >
        <img
          class="albumImg img-thumbnail"
        />
      </div>
      Top 
      Songs
    </div>
    <div
      class="topCardBody card-body"
    >
      <div
        class="card-title h5"
      >
        <div
          class="topItem"
        >
          <div
            class="number-one"
          >
            1 
          </div>
          <div
            class="number-one-item"
          >
            [loading]
             
            <span
              style="color: grey;"
            >
                by 
              [loading]
                
            </span>
          </div>
        </div>
      </div>
      <ul
        class="topListGroup list-group list-group-flush"
      >
        <li
          class="list-group-item"
        >
          <div
            style="text-align: Left;"
          >
            <span
              class="number"
            >
               
              1
            </span>
                  
            [loading]
            <span
              style="color: grey;"
            >
                 by 
              [loading]
            </span>
          </div>
        </li>
        <li
          class="list-group-item"
        >
          <div
            style="text-align: Left;"
          >
            <span
              class="number"
            >
               
              2
            </span>
                  
            [loading]
            <span
              style="color: grey;"
            >
                 by 
              [loading]
            </span>
          </div>
        </li>
        <li
          class="list-group-item"
        >
          <div
            style="text-align: Left;"
          >
            <span
              class="number"
            >
               
              3
            </span>
                  
            [loading]
            <span
              style="color: grey;"
            >
                 by 
              [loading]
            </span>
          </div>
        </li>
        <li
          class="list-group-item"
        >
          <div
            style="text-align: Left;"
          >
            <span
              class="number"
            >
               
              4
            </span>
                  
            [loading]
            <span
              style="color: grey;"
            >
                 by 
              [loading]
            </span>
          </div>
        </li>
      </ul>
      <div
        style="display: flex; justify-content: right;"
      >
        <a
          class="viewAll btn btn-primary"
          href="/all/Songs"
          role="button"
          tabindex="0"
        >
          View All
        </a>
      </div>
    </div>
  </div>
</div>
`;
